src/main/java/combat_system/Combat.java

```diff
@@ -0,0 +1,199 @@
+        if(target.getStatusEffects().get(status) == null || target.getStatusEffects().get(status)
+                < duration) {
+            target.setStatusEffect(status, duration);
+        }
+    }
+    /**
+     * Increments the round and goes through each status effect for all participants and decrements their duration
+     */
+    public void endRound() {
+        this.round += 1;
+        for (GameCharacter participant : this.participants) {
+            for(Map.Entry<StatusEffect, Integer> entry : participant.getStatusEffects().entrySet()) {
+                participant.setStatusEffect(entry.getKey(), entry.getValue()-1);
+            }
+        }
+    }
+
+    /**
+     * Used at the end of combat to clear all participants' status effects.
+     * Duration 0 causes the status to be removed, see GameCharacter.setStatusEffect for more information
+     */
+    public void clearStatus() {
+        for (GameCharacter participant : this.participants) {
+            for (Map.Entry<StatusEffect, Integer> entry : participant.getStatusEffects().entrySet()) {
+                participant.setStatusEffect(entry.getKey(), 0);
+            }
+        }
+
+    }
+
+    /**
+     * Calculates if an attack hits and the amount of damage it deals
+     * @return the string description of damage being delt
+     */
+    public String damage(int toHit, GameCharacter target, GameCharacter attacker) {
+        if(toHit >= target.getArmor().getDefense()) {
+            int health_remaining = target.getCurrentHealth()-attacker.getWeapon().getDamage();
+            target.setCurrentHealth(health_remaining);
+            if(health_remaining <= 0) {
+                if(target instanceof NonPlayerCharacter) {
+                    this.foes -= 1;
+                }
+                else {
+                    this.player_alive = false;
+                }
+            }
+
+            return "The attack hits, dealing " + attacker.getWeapon().getDamage() + " damage!";
+        }
+        else {
+            return "The attack misses!";
+        }
+    }
+
+    /**
+     * This method determines if a character is an NPC or a player and then proceeds to have them take a turn.
+     * @param participant the GameCharacter who is taking the turn
+     */
+    public void take_turn(GameCharacter participant) {
+        Random r = new Random();
+        if(participant instanceof NonPlayerCharacter) {
+            PlayerCharacter target = find_player();
+            if(r.nextBoolean()) {
+                System.out.println(participant.getName()+" enters a defensive stance");
+            }
+            else {
+                System.out.println(participant.getName()+" makes an attack against you!");
+                System.out.println(damage(r.nextInt(20), target, participant));
+            }
+        }
+        else {
+            System.out.println("It is your turn, what do you do?");
+            System.out.println("Attack");
```

Comment:
maybe move into the loop so if the player screws up many times, they will still be able to see the options

Reply:
Good idea! Changed